lib/protobuf/field/varint_field.rb

```diff
@@ -52,6 +52,7 @@ def self.encode(value, use_cache = true)
       #
 
       def acceptable?(val)
+        return true if val.is_a?(Integer) && val >= 0 && val < INT32_MAX
```

Comment:
we still want to compare to `self.class.max` right?

I understand wanting to avoid `coerce!` for performance reasons but maybe there's a better way to preserve the existing logic?

Just an idea

```ruby
def acceptable?(val)
  int_val = val.is_a?(Integer)
    val
  else
    coerce!(val)
  end

  int_val >= self.class.min && int_val <= self.class.max
end

```